src/dmclock_server.h

```diff
@@ -791,6 +798,8 @@ namespace crimson {
       {
 	assert(_erase_age >= _idle_age);
 	assert(_check_time < _idle_age);
+	// AtLimit::Reject depends on ImmediateTagCalc
+	assert(at_limit != AtLimit::Reject || !IsDelayed);
```

Comment:
wouldn't this mean using PullPQ where isDelayed is set would assert on construction if we set AtLimit?

Reply:
yes, because AtLimit::Reject is not compatible with DelayedTagCalc

in order to enforce AtLimit::Reject, do_add_request() compares the request time against RequestTag::limit - but with DelayedTagCalc, we may start with an empty RequestTag, and only calculate a real one when the request gets to the top of the `ClientRec::requests` queue